### PR TITLE
pgsql: add micrometer metrics

### LINKED ISSUE #392: Add proper Metrics using micrometer.io
motivation, see #210

### PR DESCRIPTION
**Pull request by mweirauch**
Resolves #392.

### CODE REVIEW

File: factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/PgFactStore.java [hunk 1]
```diff
@@ -65,131 +119,218 @@
...
+            time(operation, sample, exception);
+        }
+    }
+
+    private void time(@NonNull OP operation, @NonNull Sample sample, Exception e) {
+        try {
+            String exceptionTagValue = mapException(e);
+            sample.stop(timer(operation, exceptionTagValue));
+        } catch (Exception unexpected) {
+            // better safe then sorry when we got instrumentation bugs
```
Comment (uweschaefer) [inline factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/PgFactStore.java]: would it make sense to at least warn here?

Comment (mweirauch) [reply factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/PgFactStore.java]: Also scratched my head around it. In a high traffic scenario, one will spam the logs. Was thinking about replicating the `printk_once()`. On the other hand we are still in an early phase.
Ok with a `log.warn()` on every issue encounted for now?

**Comment by mweirauch** [comment]
Still investigating why my Eclipse 2019-03 doesn't format and import-order correctly according to the project specific configurations...

File: factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/PgFactStore.java (hunk 1 above)
Comment (uweschaefer) [reply factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/PgFactStore.java]: sure

**Comment by uweschaefer** [comment]
let me know if there is anything i can do to help

File: factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/PgFactStore.java [hunk 2]
```diff
@@ -65,131 +119,218 @@
     @NonNull
     final PgSubscriptionFactory subscriptionFactory;
 
-    private final FactTableWriteLock lock;
+    @NonNull
+    final FactTableWriteLock lock;
+
+    @NonNull
+    final MeterRegistry registry;
```
[the remark this answers is not in the document]
Comment (mweirauch) [reply factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/PgFactStore.java]: I assume Lomboks FieldDefaults=private and Codacy collide? But where/how the heck does Lombok chime in for the FieldDefaults?!

[the remark this answers is not in the document]
Comment (uweschaefer) [reply factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/PgFactStore.java]: lombok.config

[the remark this answers is not in the document]
Comment (mweirauch) [reply factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/internal/PgFactStore.java]: `lombok.fieldDefaults.defaultPrivate = true` from `lombok.config` applies to "_every_ field in _every_ class".

**Pull request merged**